core/templates/dev/head/pages/exploration_player/CurrentInteractionService.js

```diff
@@ -42,39 +38,37 @@ oppia.factory('CurrentInteractionService', [
       },
       registerCurrentInteraction: function(submitAnswer, validityCheckFn) {
         /**
-         * Every interaction directive should register themselves here,
-         * even those which do not use the progress nav. All arguments are
-         * optional.
+         * Each interaction directive should call registerCurrentInteraction
+         * when the interaction directive is first created.
          *
-         * @param {function} submitAnswer - Should grab the learner's answer
-         *   and pass it to onSubmit. The interaction can pass in null if
-         *   it does not use the progress nav's submit button
+         * @param {function|null} submitAnswer - Should grab the learner's
+         *   answer and pass it to onSubmit. The interaction can pass in
+         *   null if it does not use the progress nav's submit button
          *   (ex: MultipleChoiceInput).
          * @param {function} validityCheckFn - The progress nav will use this
-         *   to decide whether or not to disable the submit button. The
-         *   interaction can pass in null if it does not use the progress
-         *   nav's submit button.
+         *   to decide whether or not to disable the submit button. If the
+         *   interaction passes in null, the submit button will remain
+         *   enabled (for the entire duration of the current interaction).
          */
-        _submitAnswerFn = submitAnswer;
-        _validityCheckFn = validityCheckFn || _defaultValidityCheck;
+        _submitAnswerFn = submitAnswer || null;
+        _validityCheckFn = validityCheckFn || null;
       },
-      registerPreSubmitHook: function(hookFn) {
-        /* Register a time hook that will be called right before onSubmit.
+      registerPresubmitHook: function(hookFn) {
+        /* Register a one-time hook that will be called right before onSubmit.
```

Comment:
Ah -- "one-time" suggests that the hook is only ever run once, which isn't the case I think if multiple answers are submitted to the same interaction. I think it's fine to get rid of the word "one-time"; the next sentence clarifies it fully anyway.

Reply:
Gotcha, fixed.